### ISSUE TITLE
Couldn't find workspaceMetadata of workspace id "..." [Solution Package Generation]

### ISSUE THREAD

**Issue by ericlanteigne-semperis** [Azure/Azure-Sentinel#2730]
**Describe the bug**
Generated a package for a solution and when I verify the resource deployment I encounter an error saying it couldn't find the workspaceMetadata for the workspace.

**To Reproduce**
1. Go to 'https://portal.azure.com/?feature.IncidentTriggering=true&feature.investigationrawdata=true&feature.MultiplePlaybooksForRule=true&feature=&EntitiesFromGS=true&feature.mstimatching=true&feature.Dynamics365Connector=true&feature.Solutions=true&feature.EntityMapping=true&feature.AnalyticsEfficiency=true&feature.IncidentTimeline=true&feature.msticonnector=true&feature.mtpconnector=true&feature.BringCustomConnector=true&feature.teamscollaboration=true&feature.AlertDetails=true&feature.ZscalerPrivateAccessConnector=true&SearchKeyFieldMapping=true&feature.WindowsSecurityEventsFeature=true&feature.AdvancedSearch=true&feature.includePreviewTemplates=true&playbooksgallery=true&feature.management=true#create/Microsoft.Template'
2. Click on 'Build your own template in the editor'
3. Paste the content from mainTemplate.json into editor
4. Click on 'Save'
5. Select a Subscription, Resource Group and enter a workspace name.
6. Click on 'Review + Create'
7. Click on 'Create'
8. See error

**Expected behavior**
Expecting workspaceMetadata to be available for other resources. Not quite sure why it is not found. 

I did find in documentation a part that talks about the metadata for the workspace being stored in Cosmos DB and managed by Microsoft (https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-workspace?tabs=python#advanced). I couldn't find any other information related to workspace metadata to try to figure out why it would not be found. 

**Screenshots**
![mainTemplateThirdTry-DependenciesAdded](https://user-images.githubusercontent.com/84532420/126811984-76a057cd-0f2d-4fab-80fb-8eed9c74bf7c.png)

![workspaceInformation](https://user-images.githubusercontent.com/84532420/126812614-2f9277eb-bf0b-497d-9cc1-b48eac72fee7.png)

**Additional context**
Source files:
[1.0.2.zip](https://github.com/Azure/Azure-Sentinel/files/6870029/1.0.2.zip)

Template Validation:
![templateValidation-1 0 2](https://user-images.githubusercontent.com/84532420/126811913-67af8a01-e92c-4aa3-896f-f5c6b428d3cc.png)

**Comment by v-maudan** [comment]
@ericlanteigne-semperis, It seems Azure Sentinel is not added to your workspace. Can you please add the same and try again to redeploy the solution. let me know if you still facing any issue.[PFB screenshot]
![image](https://user-images.githubusercontent.com/75478517/126957386-ae7ee61d-ad93-4e00-96f0-596468c2dd55.png)

**Comment by v-maudan** [comment]
@ericlanteigne-semperis , Can you please check and confirm on this ?

**Comment by ericlanteigne-semperis** [comment]
@v-maudan Sorry for the delay. I was trying to find what to do yesterday afternoon but couldn't quite understand. This morning I realized after I press "+ Create" in Azure Sentinel I then get the page to add Azure Sentinel to a workspace. Once I added it to my workspace I ran the deployment again and it was successful! 
I am trying to think about how we would handle this for customers though. Because the workspace is created by the template and if this service is not added, the other resources in the template will fail. So unless we split this into a two part deployment, we would need some way to handle it within the template or during the deployment. Is there a way for the ARM template to be configured to add services to resources once they are created?

**Comment by ericlanteigne-semperis** [comment]
I am having some further issues validating the deployment although it was successful. Maybe you will have some thoughts about it. I ran deployed this into a new resource group to help isolate the resources for verification. 

Ran the deployment with the error, added Azure Sentinel to workspace, reran deployment and it was green:
![Step1](https://user-images.githubusercontent.com/84532420/127338912-c6864e80-6fc7-49ec-9c3e-d6e53fe487e7.png)
![Step2](https://user-images.githubusercontent.com/84532420/127338929-bcd0ade2-6423-4255-9b0c-501dbc0532ec.png)

Resource Group after deployment:
![ResourceGroup](https://user-images.githubusercontent.com/84532420/127338955-f27822dd-1370-4903-8d86-c062624c75e2.png)

When trying to find alert rules, none could be found although I can see the detailed information for each when selecting it from deployment page. Where would I go to view them?
![NoAlertRulesVisible](https://user-images.githubusercontent.com/84532420/127339228-d72c2425-3021-4adb-b5bd-8c8a47362510.png)
![AlertRule](https://user-images.githubusercontent.com/84532420/127339242-696d40b1-4f38-4bd0-8f8a-5a6da3be2104.png)

Data connector could also not be found, even when selecting it from the deployment page. I am a bit confused by this considering it said it was successful.
![Issue1](https://user-images.githubusercontent.com/84532420/127339398-cab8eb9d-608d-4c76-b9c7-dfc79ef55b06.png)
![OperationDetails](https://user-images.githubusercontent.com/84532420/127339895-959a6a78-dc9a-4632-9c65-5b8dbfa8a339.png)

**Comment by ericlanteigne-semperis** [comment]
I should add that once I manually added the dependsOn attributes like it mentions in documentation and I got past this issue. I wanted to bring it up though in case it can be incorporated into package generation.

**Comment by v-maudan** [comment]
@ericlanteigne-semperis,
prerequisite step before solution deployent is to have resouce group and log analytics workspace with azure sentinel enabled.
We already mentioned in the instruction for same. Don't create new resource group while solution deployment. [known issues](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/known_issues.md) 
![image](https://user-images.githubusercontent.com/75478517/127449478-1d8e8b64-2d05-4b85-aba0-d11329c1a120.png)

**Post solution deployment** 
Please check this documention
[Post solution deployment steps.docx](https://github.com/Azure/Azure-Sentinel/files/6899435/Post.solution.deployment.steps.docx)

Please let me know if you still facing any issue.Thanks!

**Comment by ericlanteigne-semperis** [comment]
@v-maudan Thanks for the explanation and reference. I was able to find my resources that were deployed. 
We can close this issue since I am not blocked by this anymore. Thank you
